q2_diversity_lib/_util.py

```diff
@@ -22,17 +31,79 @@ def _drop_undefined_samples(counts: np.ndarray, sample_ids: np.ndarray,
     return (filtered_counts, filtered_sample_ids)
 
 
-def _disallow_empty_tables(some_function):
+def _disallow_empty_tables_passed_object(some_function):
+    @wraps(some_function)
+    def wrapper(*args, **kwargs):
+        try:
+            bound_signature = signature(some_function).bind(*args, **kwargs)
+            table = bound_signature.arguments['table']
+        except KeyError as ex:
+            raise TypeError("The wrapped function has no parameter "
+                            + str(ex) + ".")
+        if type(table) != bTable:
+            raise TypeError("This method requires a biom.Table object "
+                            "and received a String")
+        if table.is_empty():
+            raise ValueError("The provided table object is empty")
+        return some_function(*args, **kwargs)
+    return wrapper
+
+
+# TODO: Passing filepaths appears to be blowing up biom - load call on l52
```

Comment:
Based on my read of the code, and some hands-on testing - this is due to the framework supplying the registered view type of `BIOMV210Format`, but this util, and most or all of your unit tests are testing with a _different_ view type - a `str` filepath.